homeassistant/helpers/entity_registry.py

```diff
@@ -923,23 +923,20 @@ async def async_migrate_entries(
             ent_reg.async_update_entity(entry.entity_id, **updates)
 
 
+@callback
+def async_resolve_entity(registry: EntityRegistry, entity_id_or_uuid: str) -> str:
+    """Resolve an entity id or UUID to an entity id or None."""
+    if valid_entity_id(entity_id_or_uuid):
+        return entity_id_or_uuid
+    if (entry := registry.entities.get_entry(entity_id_or_uuid)) is None:
+        raise vol.Invalid(f"Unknown entity registry entry {entity_id_or_uuid}")
```

Comment:
the doc string says this returns None which makes sense because that's how other get methods work. 
```suggestion
        return None
```

Reply:
I'd rather rename the function and update the doc string:
```py
def async_validate_entity_id(entity_id_or_uuid: str) -> str | None:
    """Resolve an entity id or UUID to an entity id.
    Raises vol.Invalid if the entity or UUID is invalid, or if the UUID is not
    associated with an entity registry item.
    """
```